database/src/migrations/20240315000000_update_funding_source_table.ts

```diff
@@ -0,0 +1,32 @@
+import { Knex } from 'knex';
+
+/**
+ * Added unique key constraint to survey funding source table
+ *
+ * @export
+ * @param {Knex} knex
+ * @return {*}  {Promise<void>}
+ */
+export async function up(knex: Knex): Promise<void> {
+  await knex.raw(`--sql
+
+    SET SEARCH_PATH=biohub;
+
+    ----------------------------------------------------------------------------------------
+    -- Add unique key constraints
+    ----------------------------------------------------------------------------------------
+    CREATE UNIQUE INDEX survey_funding_source_uk1 ON survey_funding_source(funding_source_id, survey_id);
```

Comment:
Technically we would need to account for possible duplicate funding data in prod, or else this would fail.

That being said, I checked prod, and there doesn't seem to be any duplicate funding sources, so its probably fine for now.